mantidproject/mantid
# Advanced plotting menu in Workbench

**Description of work.**
This PR adds the advanced plotting menu present in MantidPlot to Workbench. 

![image](https://user-images.githubusercontent.com/52415735/76956750-8a5b8880-690c-11ea-9d49-27567be4c898.png)

**To test:**
When you right-click a workspace and go to "Plot", there should now be an "Advanced..." option. This should open the Plot Spectrum dialog but with the following additional options:
- A check box for enabling error bars.
- Choosing the log value to plot against. For 1D plots this will just change the label of each line in the legend to the value of the log you have selected.
- If you open the dialog with three or more workspaces selected, Surface and Contour become available options in the plot type combo box. The log value selected changes one of the axes on the plot, and you can also change what the axis label is.

Currently surface plots use the same figure options as 1D and 2D plots which means only the x and y axis limits can be changed. The options to change axis scale are disabled because they don't work correctly for 3D plots at the moment.

For testing I have been using the loqdemo files from the ISIS sample data because these workspaces have a lot of sample logs. If you open the Advanced Plots menu with two or more workspaces selected, the log value combo box should contain every sample log that is present in all workspaces and has a numeric value. For single workspaces you should just get "Workspace name" and "Custom". Custom log values need to be a distinct list of floats in numerical order, with the number of values being equal to the number of workspaces for surface and contour plots, or the number of curves for the other plot types. if you enter an invalid input you can hover over the red asterisk to find out what's wrong. For surface and contour plots you can only enter a single spectrum number or workspace index. 

Fixes #27450 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
